File: problems/2032_two_out_of_three/Solution.java
import java.util.*;

class Solution {
    public List<Integer> twoOutOfThree(int[] nums1, int[] nums2, int[] nums3) {
        Set<Integer> s1 = new HashSet<>(), s2 = new HashSet<>(), s3 = new HashSet<>();
        for (int n : nums1) s1.add(n);
        for (int n : nums2) s2.add(n);
        for (int n : nums3) s3.add(n);
        Set<Integer> result = new HashSet<>();
        for (int n : s1) if (s2.contains(n) || s3.contains(n)) result.add(n);
        for (int n : s2) if (s3.contains(n)) result.add(n);
        return new ArrayList<>(result);
    }
}

File: problems/2032_two_out_of_three/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    vector<int> twoOutOfThree(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3) {
        unordered_set<int> s1(nums1.begin(), nums1.end());
        unordered_set<int> s2(nums2.begin(), nums2.end());
        unordered_set<int> s3(nums3.begin(), nums3.end());
        unordered_set<int> result;
        for (int n : s1) if (s2.count(n) || s3.count(n)) result.insert(n);
        for (int n : s2) if (s3.count(n)) result.insert(n);
        return vector<int>(result.begin(), result.end());
    }
};
